source/extensions/path/uri_template_lib/uri_template_internal.cc

```diff
@@ -44,7 +44,7 @@ constexpr absl::string_view kLiteral = "a-zA-Z0-9-._~" // Unreserved
                                        "%"             // pct-encoded
                                        "!$&'()+,;"     // sub-delims excluding *=
                                        ":@"
-                                       "="; // user included "=" allowed
```

Comment:
I do not know why * was excluded as a match character. But for a change like this a runtime guard is required as a minimum, since it changes the existing behavior of the matcher.

Reply:
Hi @yanavlasov, thanks for looking into this, I have moved the feature behind a runtime guard.